plugin/evm/vm.go

```diff
@@ -998,7 +999,18 @@ func (vm *VM) CreateHandlers(context.Context) (map[string]http.Handler, error) {
 	}
 
 	if vm.config.WarpAPIEnabled {
-		validatorsState := warpValidators.NewState(vm.ctx)
+		requirePrimaryNetworkSigners := func() bool {
```

Comment:
I wonder if we should specifically enable this at EUpgrade. It does not seem required since one needs to disable/re-enable the precompile with this config (so basically they will perform upgrade), but wanted to get a confirmation.

Reply:
I don't think it should be, since SoV network is an opt-in change then configuring warp precompile should be also done as part of that opt-in.
I think we should just test both types of activation in some e2e test (opt-in and default).